### PR TITLE
MTU settings are now L3 rather than L2 based.

### PR DESCRIPTION
**Pull request by kbara**
IE, an IPv6 MTU of 1500 is now 1500 bytes for the IPv6 header and payload,
rather than 1500 - (ethernet + possibly vlan) bytes.
All tests are updated accordingly and pass.
This also corrects the result packet for replying to attempt to fragment a
DF packet with a vlan: it's now 576 rather than 580 bytes, in accordance
with specifications on ICMP packet size.

### CODE REVIEW

**Comment by kbara** [comment]
I believe this should fully resolve https://github.com/Igalia/snabbswitch/issues/100

File: src/apps/lwaftr/fragmentv6.lua [hunk 1]
```diff
@@ -173,9 +173,12 @@ end
 -- Packets have two parts: an 'unfragmentable' set of headers, and a
 -- fragmentable payload.
 function fragment_ipv6(ipv6_pkt, unfrag_header_size, l2_size, mtu)
-   if ipv6_pkt.length <= mtu then
+   if ipv6_pkt.length - l2_size <= mtu then
       return ipv6_pkt -- No fragmentation needed
    end
+   -- Hack: continue to use the previous code, even though MTU is now defined
+   -- as an l3 rather than l2 size.
```
Comment (andywingo) [inline src/apps/lwaftr/fragmentv6.lua]: can we just rename to mtu_with_l2 or something?

File: src/apps/lwaftr/fragmentv4.lua [hunk 2]
```diff
@@ -56,9 +56,10 @@ FRAGMENT_FORBIDDEN = 3
 --     an ICMP Datagram Too Big (Type 3, Code 4) packet back to the sender.
 --
 function fragment_ipv4(ipv4_pkt, l2_size, mtu)
-   if ipv4_pkt.length <= mtu then
+   if ipv4_pkt.length - l2_size <= mtu then
       return FRAGMENT_UNNEEDED, ipv4_pkt
    end
+   mtu = mtu + l2_size
```
Comment (andywingo) [inline src/apps/lwaftr/fragmentv4.lua]: can we rename to mtu_with_l2_header or something please, just to help future readers? :)

**Comment by andywingo** [comment]
LGTM with just a couple nits.  Great stuff, thank you!

**Comment by aperezdc** [comment]
LGTM.

**Comment by andywingo** [comment]
Merging without nits fixed; please fix nits on lwaftr_olive in a followup.  Thank you again :-))

**Pull request merged**